File: matlab/read_write/get_species.m
function		s = get_species(initial)

%		s = get_species(initial)

s = [] ;
S = read_csv('species.csv') ;
if nargin==0,
	for k=1:length(S),
		fprintf(' %s %s\n',S(k).Initial,S(k).Common_name) ;
	end	
	return
end

% look for S.Initial that matches species initial
k = strmatch(initial,{S.Initial},'exact') ;

if isempty(k),
	fprintf(' No entry matching "%s" in species file - edit file and retry\n',initial) ;
	return ;
end

if length(k)>1,
	fprintf(' Multiple entries matching "%s" in species file:\n',initial) ;
	for kk=1:length(k),
		fprintf(' %d %s\n',kk,S(k(kk)).Common_name) ;
	end
	n = input(' Enter number for correct species... ','s') ;
	n = str2double(n) ;
	if isempty(n) || isnan(n) || n<1 || n>length(k), return ; end
	k = k(n) ;
end

s = S(k) ;

File: R/tagtools/R/get_species.R
#' Find matching species in a list of marine mammals
#' 
#' @param initial a two-letter code for the species of interest (first letter of Genus and first letter of species)
#' @noMd

get_species <- function(initial){
  S <- utils::read.csv(system.file('extdata', 'species.csv', package='tagtools'),
                       stringsAsFactors = FALSE)

  if (missing(initial)){
    print(S)
    return()
  }
  
  # look for S.Initial that matches species initial
  k <- initial == S$Initial
  if (sum(k) == 0){
    stop(sprintf(' No entry matching "%s" in species file - edit file and retry\n',initial))
  }
  
  if (sum(k) > 1){
    matches <- S[k,]
    print(matches)
    r <- readline(prompt = 'Type the row number of the correct match: ')
    r <- as.numeric(r)
    if (length(n) == 0 | is.na(n) | n < 1 | n > nrow(matches)){
      stop('User input not recognized.')
    }
    k <- matches[r,'Common_name'] == S$common_name
  }
  return(S[k,])
  }